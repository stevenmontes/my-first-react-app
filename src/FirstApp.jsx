const getMessage = (name) => `Hola ${name}`;

const FirstApp = () => {
    return (
        <>
            <h1>{getMessage('Chris')}</h1>
            <p>Soy un subtitulo</p>
        </>
    )
}

export default FirstApp;
