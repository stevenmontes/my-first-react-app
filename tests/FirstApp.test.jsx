import { render } from "@testing-library/react";
import FirstApp from "../src/FirstApp";

describe('Pruebas en FirstApp', () => {
    test('should match with the snapshot', () => {
        const { container } = render(<FirstApp title="Hola, Soy Goku" />);

        expect(container).toMatchSnapshot();
    });

    test('should show a title in h1', () => {
        const title = "Hola, Soy Goku";
        const { container, getByText } = render(<FirstApp title={title} />);

        expect(getByText(title)).toBeTruthy();

        const h1 = container.querySelector('h1');
        expect(h1.innerHTML).toContain(title);

    })
});