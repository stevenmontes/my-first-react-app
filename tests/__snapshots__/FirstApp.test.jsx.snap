// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp should match with the snapshot 1`] = `
<div>
  <h1>
    Hola, Soy Goku
  </h1>
  <p>
    No hay subtitulo
  </p>
  <p>
    Chris Montes
  </p>
</div>
`;
